Go: Do not track taint into a `sync.Map` via the key of a key-value pair (#18920)
github/codeql: go/ql/lib/change-notes/2025-03-04-improve-models-for-sync-map.md

---
category: minorAnalysis
---

> * We no longer track taint into a `sync.Map` via the key of a key-value pair, since there is no way of reading the keys stored in the `sync.Map` back.

Reviewer: Regardless of whether we model `Range` or not right now, we should edit this to accurately reflect that there is at least one way to get the keys out of a `sync.Map`, but that we don't model that.

```suggestion
* We no longer track taint into a `sync.Map` via the key of a key-value pair, since we do not model any way in which keys can be read from a `sync.Map`.
```
